aiogram/types/chat.py

```diff
@@ -63,6 +63,16 @@ def user_url(self):
 
         return f"tg://user?id={self.id}"
 
+    @property
+    def shifted_id(self) -> int:
+        """
+        Get shifted id of chat, e.g. for private links
+
+        For example: -1001122334455 -> 1122334455
+        """
+        shift = -1000000000000
+        return shift - self.id
```

Comment:
Private chats is also should have shifted id?

Reply:
Yes, you can double-click (or long tap) on message and click "Copy link"

Reply:
Hmm, I thought you were talking about closed chats. So private chat with bot doesnt have shifted_id, I will add check for it.

Reply:
I'm sure you can use `-1_000_000_000_000` for readability without breaking anything.

Reply:
Cool, but for what? It is internal numbering of Telegram, not for calculation or anything else